src/qibocal/protocols/utils.py

```diff
@@ -757,39 +762,248 @@ def table_html(data: dict) -> str:
     )
 
 
-def extract_feature(
-    x: np.ndarray,
-    y: np.ndarray,
-    z: np.ndarray,
-    find_min: bool,
-):
-    """Extract feature using confidence intervals.
+def euclidean_metric(point1: np.ndarray, point2: np.ndarray):
+    """Euclidean distance between two arrays."""
+    return np.linalg.norm(point1 - point2)
+
+
+def zca_whiten(X):
+    """
+    Applies ZCA whitening to the data (X)
+    https://en.wikipedia.org/wiki/Whitening_transformation
 
-    Given a dataset of the form (x, y, z) where a spike or a valley is expected,
-    this function discriminate the points (x, y) with a signal, from the pure noise
-    and return the first ones.
+    X: numpy 2d array
+        input data, rows are data points, columns are features
+
+    Returns: ZCA whitened 2d array
     """
+    assert X.ndim == 2
+    EPS = 10e-5
+
+    #   covariance matrix
+    cov = np.dot(X.T, X)
+    #   d = (lambda1, lambda2, ..., lambdaN)
+    d, E = np.linalg.eigh(cov)
+    #   D = diag(d) ^ (-1/2)
+    D = np.diag(1.0 / np.sqrt(d + EPS))
+    #   W_zca = E * D * E.T
+    W = np.dot(np.dot(E, D), E.T)
+
+    X_white = np.dot(X, W)
+
+    return X_white
+
+
+def custom_filter_mask(matrix_z: np.ndarray):
+    """Applying a mask compsosed by first a ZCA transformation and then a gaussian filter with variance 1."""
+
+    zca_z = zca_whiten(matrix_z)
+    # adding zca filter for filtering out background noise gradient
+    zca_gauss_z = ndimage.gaussian_filter(zca_z, 1)
+    # adding gaussia fliter with unitary variance for blurring the signal and reducing noise
+
+    return zca_gauss_z
+
+
+def scaling_global(sig: np.ndarray) -> np.ndarray:
+    """Min–max scaling over the whole np.ndarray (global)."""
+    sig_min = np.min(sig)
+    return (sig - sig_min) / (np.max(sig) - sig_min)
+
+
+def scaling_slice(sig: np.ndarray, axis: int) -> np.ndarray:
```

Comment:
```suggestion
def scaling_slice(sig: np.ndarray, axis: Optional[int]) -> np.ndarray:
```

Let's not restrict the range of options the NumPy accepts ;)